src/Microsoft.AspNetCore.Routing/Matching/HttpMethodMatcherPolicy.cs

```diff
@@ -261,6 +266,19 @@ private Endpoint CreateRejectionEndpoint(IEnumerable<string> httpMethods)
                 Http405EndpointDisplayName);
         }
 
+        public static bool ContainsHttpMethod(List<string> httpMethods, string httpMethod)
+        {
+            for (int i = 0; i < httpMethods.Count; i++)
+            {
+                if (StringComparer.OrdinalIgnoreCase.Equals(httpMethods[i], httpMethod))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
```

Comment:
https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.list-1.binarysearch?view=netframework-4.7.2

Reply:
That would require the list to be sorted, and this is used while building the list, so it would either need to be resorted after each addition, or added to the right place in the list. I'm not sure it is worth the complexity for what will always be a small list.

Reply:
It just makes me a little sad to see somethign like this in in the BCL :(

Reply:
Btw if you want a really nice trick for this... https://github.com/aspnet/Mvc/blob/a67d9363e22be8ef63a1a62539991e1da3a6e30e/src/Microsoft.AspNetCore.Mvc.Core/Formatters/TextOutputFormatter.cs#L251

List.BinarySearch is best API ever. I noticed that you are sorting `allHttpMethods` eventually anyway.